### DISCUSSION TITLE
Show accessible fields before methods in completion menu

### DISCUSSION THREAD

**Discussion by boxbeam** [helix-editor/helix#10944] [Troubleshooting]
When typing `.` after a struct value in rust (with rust-analyzer LSP), you're shown a listing of all methods and fields. The fields are intermixed with the methods, leading me to usually have to scroll around a lot of (mostly derived) methods to find the fields I'm looking for. In many cases it ends up being faster to just navigate to the struct and see its fields, then type out the name of the one I need. It would be really nice if accessible struct fields would always show above methods.

In the same vein, it would be really nice if direct-impl methods would show above trait methods.

**Comment by GNUSheep** [comment]
What version of rust-analyzer do u use?

Because for me it's working well
![image](https://github.com/helix-editor/helix/assets/54682807/f4056518-4aab-4cc2-b593-a8b638708f97)
![image](https://github.com/helix-editor/helix/assets/54682807/09f87c31-8381-4875-ab11-8f06be66e21f)

**Comment by boxbeam** [comment]
That happened to me for a day then stopped, it was strange. I'll have to try updating.
